webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -178,21 +179,24 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
     if (report.validSubject) {
       return report.subject.value;
     }
-    if (report.subject.name) {
+    if (report.subject?.name) {
       return report.subject.name;
     }
     return this.translateService.instant('report.subject.unknown');
   }
 
-  private prepareReports(reports) {
+  prepareReports(reports) {
     return reports.map(report => {
       const form = _find(this.forms, { code: report.form });
       report.icon = form && form.icon;
       report.heading = this.getReportHeading(form, report);
       report.summary = form ? form.title : report.form;
       report.lineage = report.subject && report.subject.lineage || report.lineage;
+      // remove the lineage level that belongs to the offline logged-in user, normally the last one
+      if (this.currentLevel) {
+        report.lineage = report.lineage.filter(level => level !== this.currentLevel);
```

Comment:
From my investigation it can only be in the middle of lineage, when the user is online only, and this case, we decide to show the whole lineage without altering it